frontend/cypress/e2e/talentsearch/search-workflows.cy.js

```diff
@@ -8,89 +11,308 @@ describe("Talent Search Workflow Tests", () => {
       aliasMutation(req, "createPoolCandidateSearchRequest");
     });
 
-    cy.visit("/en/search");
+    // select some dimensions to use for testing
+    cy.getSkills().then((allSkills) => {
+      cy.wrap(allSkills[0]).as("testSkill"); // take the first skill for testing
+    });
+    cy.getGenericJobTitles().then((allGenericJobTitles) => {
+      const testGenericJobTitle1 = allGenericJobTitles[0]; // take the first ID for testing matching
+      cy.wrap(testGenericJobTitle1).as("testGenericJobTitle1");
+      cy.wrap(testGenericJobTitle1.classification).as("testClassification1");
+
+      const testGenericJobTitle2 = allGenericJobTitles[1]; // take the second ID for testing rejection
+      cy.wrap(testGenericJobTitle2.id).as("testGenericJobTitleId2");
+      cy.wrap(testGenericJobTitle2.classification).as("testClassification2");
+    });
   });
 
-  const searchReturnsGreaterThanZeroApplicants = () => {
-    cy.findByRole("heading", {
-      name: /Results: [1-9][0-9]* matching candidate/i,
+  // calculate how the classification will be displayed in the UI
+  const classificationName = (classification) =>
+    `${classification.group}-${classification.level
+      .toString()
+      .padStart(2, "0")}`;
+
+  it("searches for a candidate with all the filters and submits a request", () => {
+    // have a unique ID to track the test objects with
+    const uniqueTestId = Date.now().valueOf();
+    cy.log(`Test run ${uniqueTestId}`);
+
+    const searchFindsMySingleCandidate = () => {
+      cy.findAllByRole("article", {
+        name: `Cypress Test Pool EN 1 ${uniqueTestId} (IT-01 Business Line Advisory Services)`,
+      }).within(() => {
+        cy.contains("There is 1 matching candidate in this pool");
+
+        cy.findByRole("button", { name: /Request Candidates/i })
+          .should("exist")
+          .and("be.visible")
+          .and("not.be.disabled");
+      });
+    };
+
+    const searchRejectsMySingleCandidate = () => {
+      cy.findAllByRole("article", {
```

Comment:
findAllByRole can return many, and within only works for one:

https://docs.cypress.io/api/commands/within#Requirements

> ## Requirements 
>
> `.within()` requires being chained off a previous command that yields exactly one DOM element.

```suggestion
      cy.findByRole("article", {
```

Confirmed this works :)

Reply:
Good catch!  I've changed this to findByRole in https://github.com/GCTC-NTGC/gc-digital-talent/pull/4998/commits/44f59935d54487c1cab50f88ab2dd90f738fdcc8.